packages/wallets/src/features/cashier/modules/Transfer/hooks/useTransferMessages/utils/lifetimeAccountLimitsBetweenWalletsMessageFn.ts

```diff
@@ -0,0 +1,91 @@
+import { TMessageFnProps } from '../types';
+
+// this function should work once BE WALL-1440 is delivered
+const lifetimeAccountLimitsBetweenWalletsMessageFn = ({
+    activeWallet,
+    displayMoney,
+    exchangeRates,
+    limits,
+    sourceAccount,
+    sourceAmount,
+    targetAccount,
+}: TMessageFnProps) => {
+    if (sourceAccount?.account_category !== 'wallet' || targetAccount?.account_category !== 'wallet') return null;
+
+    const sourceWalletType = sourceAccount.account_type === 'crypto' ? 'crypto' : 'fiat';
+    const targetWalletType = targetAccount.account_type === 'crypto' ? 'crypto' : 'fiat';
+    const limitsCaseKey = `${sourceWalletType}_to_${targetWalletType}` as const;
+
+    //@ts-expect-error needs backend type
+    const allowedSumActiveWalletCurrency = limits?.lifetime_transfers?.[limitsCaseKey].allowed as number;
+    //@ts-expect-error needs backend type
+    const availableSumActiveWalletCurrency = limits?.lifetime_transfers?.[limitsCaseKey].available as number;
+
+    if (
+        !sourceAccount.currency ||
+        !exchangeRates?.rates?.[sourceAccount.currency] ||
+        !targetAccount.currency ||
+        !exchangeRates?.rates?.[targetAccount.currency] ||
+        !sourceAccount.currencyConfig ||
+        !targetAccount.currencyConfig
+    )
+        return null;
+
+    const transferDirection = activeWallet.loginid === sourceAccount.loginid ? 'from' : 'to';
+
+    const allowedSumConverted =
+        allowedSumActiveWalletCurrency *
+        (exchangeRates?.rates[transferDirection === 'from' ? targetAccount.currency : sourceAccount.currency] ?? 1);
+    const availableSumConverted =
+        availableSumActiveWalletCurrency *
+        (exchangeRates?.rates[transferDirection === 'from' ? targetAccount.currency : sourceAccount.currency] ?? 1);
+
+    const sourceCurrencyLimit = transferDirection === 'from' ? allowedSumActiveWalletCurrency : allowedSumConverted;
+    const targetCurrencyLimit = transferDirection === 'from' ? allowedSumConverted : allowedSumActiveWalletCurrency;
+
+    const sourceCurrencyRemainder =
+        transferDirection === 'from' ? availableSumActiveWalletCurrency : availableSumConverted;
+    const targetCurrencyRemainder =
+        transferDirection === 'from' ? availableSumConverted : availableSumActiveWalletCurrency;
+
+    const formattedSourceCurrencyLimit = displayMoney?.(
+        sourceCurrencyLimit,
+        sourceAccount.currencyConfig.display_code,
+        sourceAccount.currencyConfig.fractional_digits
+    );
+    const formattedTargetCurrencyLimit = displayMoney?.(
+        targetCurrencyLimit,
+        targetAccount.currencyConfig.display_code,
+        targetAccount.currencyConfig?.fractional_digits
```

Comment:
Why is the optional chaining operator (?) only added on certain endpoints? Is there no possibility for the other endpoints to be undefined?

Reply:
Hi @lubega-deriv,
I am not sure about this, maybe @rostislav-deriv can help.

Reply:
Hi @lubega-deriv,
This is only for the places which use data related to the destination account in transfer. When the user opens the  transfer tab, source of transfer is pre-selected as the authorized wallet but the destination of transfer is yet to be selected by the user. Hence, those places which use the destination account's data will throw error.
I hope my point is clear 😓.